FGB-insite-team-task/src/app/service/fgb-detail.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from "@angular/common/http";
+import {Game} from '../fgb-models/game.model';
+import { Season } from '../fgb-models/season.model';
+import { Product } from '../fgb-models/product.model';
+import { Attendance } from '../fgb-models/attendance.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FgbDetailService {
+  gameList:Game[];
+  seasonList:Season[];
+  productList:Product[];
+  attendanceList:Attendance[];
+  gameNumber:number;
+  seasonNumber:number;
+  productNumber:string;
+
+  readonly rootURL = 'https://localhost:5001/api';
+
+  constructor(private http: HttpClient) {
+  }
+
+  getGame() {
+    this.http.get<Game[]>(this.rootURL + '/game')
+    .toPromise()
+    .then(res => this.gameList = res as Game[]);
```

Comment:
Converting these Observables to Promises is really not necessary, and neither is the use of the `as` casting (using `get<Game[]>` ensures the HttpClient will only return that type). You'd be able to accomplish the same thing with:
```
this.http.get<Game[]>(this.rootURL + '/game')
    .subscribe(games => this.gameList = games);
```

Reply:
Again though, this function should return the observable so that whatever class is calling it can manipulate the data and handle it as it pleases:
```
getGames() {
    return this.http.get<Game[]>(this.rootUrl + '/game/);
}
```